### ISSUE TITLE
Route `script` should be optional/nullable

### ISSUE THREAD

**Issue by signalnerve** [cloudflare/wrangler#123]
The current version of the `Route` impl says that `script` is a required string:

```rust
pub struct Route {
  script: String,
  pattern: String,
}
```

https://github.com/cloudflare/wrangler/blob/master/src/commands/publish/route.rs#L8-L11

When deploying a route as part of `wrangler publish`, if a user's Cloudflare account has any routes published _without_ an attached script, publishing will fail:

```
assets-on-workers $ RUST_BACKTRACE=1 wrangler publish
⚠️ JavaScript project found. Skipping unecessary build!
🥳 Successfully published your script.
Error: Error("invalid type: null, expected a string", line: 21, column: 20)

stack backtrace:
   0: failure::backtrace::internal::InternalBacktrace::new::h899423eefab5fdf4 (0x10c752d98)
   1: <failure::backtrace::Backtrace as core::default::Default>::default::h2619c04ef0e2f942 (0x10c752f94)
   2: wrangler::commands::publish::route::Route::publish::h28226564daaa540a (0x10c40d249)
   3: wrangler::commands::publish::publish::h8e72dac64475e3e4 (0x10c4391bc)
   4: wrangler::main::hea7dc5c1a0e03c38 (0x10c405119)
   5: std::rt::lang_start::{{closure}}::h768c74ced068225e (0x10c401a5a)
   6: std::panicking::try::do_call::hf09ae321d60b9a09 (0x10c7cd587)
   7: ___rust_maybe_catch_panic (0x10c7d738e)
   8: std::rt::lang_start_internal::h5162fc1065feec16 (0x10c7ce06d)
   9: _main (0x10c405748)
```

The `script` field on a `Route` should be optional so that the code can safely iterate through each route returned from the API, and check for an existing route during publish.

**Comment by signalnerve** [comment]
Turns out that this might be happening with non-webpack projects, too. I'm deploying a different worker with project type `javascript` and getting this error:

```
assets-on-workers $ RUST_BACKTRACE=1 wrangler publish
⚠️ JavaScript project found. Skipping unecessary build!
🥳 Successfully published your script.
Error: Error("invalid type: null, expected a string", line: 21, column: 20)

stack backtrace:
   0: failure::backtrace::internal::InternalBacktrace::new::h899423eefab5fdf4 (0x10c752d98)
   1: <failure::backtrace::Backtrace as core::default::Default>::default::h2619c04ef0e2f942 (0x10c752f94)
   2: wrangler::commands::publish::route::Route::publish::h28226564daaa540a (0x10c40d249)
   3: wrangler::commands::publish::publish::h8e72dac64475e3e4 (0x10c4391bc)
   4: wrangler::main::hea7dc5c1a0e03c38 (0x10c405119)
   5: std::rt::lang_start::{{closure}}::h768c74ced068225e (0x10c401a5a)
   6: std::panicking::try::do_call::hf09ae321d60b9a09 (0x10c7cd587)
   7: ___rust_maybe_catch_panic (0x10c7d738e)
   8: std::rt::lang_start_internal::h5162fc1065feec16 (0x10c7ce06d)
   9: _main (0x10c405748)
```

And my `wrangler.toml`:

```toml
name = "hugo-workers"
account_id = "<redacted>"
zone_id = "<redacted>"
route = "https://hugo-workers.signalnerve.com/*"
type = "javascript"
```

**Comment by ashleygwilliams** [comment]
closed by #134
